database/src/models/account.rs

```diff
@@ -253,3 +266,58 @@ impl<'a, 'r> FromRequest<'a, 'r> for Account {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::env;
+
+    #[test]
+    fn it_calculates_domain_correctly() {
+        let local_domain: &str = "local.domain";
+        let test_env_domain = env::var("DOMAIN");
+        env::set_var("DOMAIN", local_domain);
+        let account_with_null_domain: Account = Account {
+            id: 1,
+            uri: Some("http://fake.value/1".to_string()),
+            domain: None,
+            username: "account1".to_string(),
+            display_name: None,
+            summary: None,
+        };
+        let account_with_local_domain: Account = Account {
+            id: 2,
+            uri: Some("http://fake.value/2".to_string()),
+            domain: Some(LOCAL_ACCOUNT_DOMAIN.to_string()),
+            username: "account2".to_string(),
+            display_name: None,
+            summary: None,
+        };
+        let account_with_remote_domain: Account = Account {
+            id: 2,
+            uri: Some("http://fake.value/3".to_string()),
+            domain: Some("fake.value".to_string()),
+            username: "account3".to_string(),
+            display_name: None,
+            summary: None,
+        };
+        assert_eq!(
+            account_with_null_domain.get_domain(),
+            local_domain,
+            "Account with NULL domain should be calculated as the local domain"
+        );
+        assert_eq!(
+            account_with_local_domain.get_domain(),
+            local_domain,
+            "Account created locally should be calculated as the local domain"
+        );
+        assert_eq!(
+            account_with_remote_domain.get_domain(),
+            "fake.value",
+            "Account from remote should be calculated as their origin"
+        );
+        if let Ok(original_domain) = test_env_domain {
+            env::set_var("DOMAIN", original_domain);
```

Comment:
I'm not sure if this is correct or not wrt how Rust isolates test runs, this may just be silly housekeeping for no benefit.

Reply:
I think I'd likely redesign the `get_domain` path / how `DOMAIN` is acquired before I went too far down that path, nothing easy or simple is occurring to me in terms of how to do that. Like `get_domain()` defers to `get_domain_with_default` which allows for injecting an outside value, such that the tests can exercise the behavior without worrying about global state.

That said, this is a pretty trite unit test, so it could also just be removed and then the concern removed as well. idk, not super attached to any path here, so likely to defer to anyone who has strong opinions on what's right.